src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -134,8 +139,9 @@ private static ParquetProperties createParquetProperties() {
         // pages is the same.
         // The quick fix is to effectively disable the page size/row limit
         // to always have one page per chunk until server side is generalised.
-        .withPageSize((int) Constants.MAX_CHUNK_SIZE_IN_BYTES * 2)
+        .withPageSize(pageSizeLimit)
         .withPageRowCountLimit(Integer.MAX_VALUE)
+        .withMinRowCountForPageSizeCheck(Integer.MAX_VALUE)
```

Comment:
I assume this is all we need? If yes, do we still need the estimatedUncompressedChunkSizeInBytes?

Reply:
Also let's add comment on why we need to do it and a TODO to update it later

Reply:
> I assume this is all we need? If yes, do we still need the estimatedUncompressedChunkSizeInBytes?

Yes. The idea was to estimate size after compression (pageSizeLimit for Parquet)) based on size of data before compression (estimatedUncompressedChunkSizeInBytes). Since we don't have a good way to estimate it atm, I can remove the estimatedUncompressedChunkSizeInBytes for now.

Reply:
No strong preference here since the `MAX_CHUNK_SIZE_IN_BYTES * 2` doesn't make much sense already :) I think the safest option is set the limit to be MAX_BLOB_SIZE_IN_BYTES

Reply:
I would agree with `MAX_BLOB_SIZE_IN_BYTES`, but the problem is that we saw cases when this limits are used to allocate buffers in mem taking this size as initial or for estimates. We risk to overallocate too much memory with such conservative limits imo.

Reply:
I removed the estimate because we don't even need to set `withPageSize(limit)`. The line I added `.withMinRowCountForPageSizeCheck(Integer.MAX_VALUE)` will disable both checks (1) when to flush to rowgroup and then (2) when to flush column data to pages. The page size limit we set doesn't have any effect on the decision when to flush, it just allocates memory with that initial size.